save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -105,23 +107,26 @@ internal fun String.getLineNumber(warningRegex: Regex,
                                   file: Path?,
                                   linesFile: List<String>?,
 ): Int? {
-    val groups = warningRegex.find(this)?.groups ?: return null
+    if (lineGroupIdx == null) {
+        // line capture group is not configured in save.toml
+        return null
+    }
 
-    return lineGroupIdx?.let {
-        val lineValue = groups[lineGroupIdx.toInt()]!!.value
-        if (lineValue.isEmpty() && lineNum != null && linesFile != null) {
-            return plusLine(file, warningRegex, linesFile, lineNum)
-        } else {
-            lineValue.toIntOrNull() ?: run {
-                if (lineValue[0] != placeholder[0]) {
-                    throw ResourceFormatException("The group <$lineValue> is neither a number nor a placeholder.")
-                }
-                try {
-                    val line = lineValue.substringAfterLast(placeholder)
-                    lineNum!! + 1 + if (line.isNotEmpty()) line.toInt() else 0
-                } catch (e: Exception) {
-                    throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.describe()}")
-                }
+    val groups = warningRegex.find(this)?.groups ?: return null
+    val lineValue = groups[lineGroupIdx.toInt()]!!.value
+    return if (lineValue.isEmpty() && lineNum != null && linesFile != null) {
+        nextLineNotMatchingRegex(file!!, warningRegex, linesFile, lineNum)
+    } else {
+        lineValue.toIntOrNull() ?: run {
+            if (lineValue[0] != placeholder[0]) {
+                throw ResourceFormatException("The group <$lineValue> is neither a number nor a placeholder.")
+            }
+            try {
+                val line = lineValue.substringAfterLast(placeholder)
+                nextLineNotMatchingRegex(file!!, warningRegex, linesFile!!, lineNum!!) +
+                        if (line.isNotEmpty()) line.toInt() else 0
+            } catch (e: Exception) {
```

Comment:
Yes, here is probably can be only `NumberFormatException`, but to be safe and to not crash because of some weird and unexpected errors, I think it makes sense to catch everything during parsing